Fix clustermesh-apiserver dependencies on pkg/option

tgraf opened the pull request:
The clustermesh-apiserver code indirectly used `pkg/option` without initializing it properly. Instead of fixing the initialization, refactor the dependency code to no longer use `pkg/option` but offer interfaces to pass in the required configuration.

``` release-note
Fix clustermesh-apiserver to respect non-standard cluster names and support `--k8s-service-proxy-name` option.
```

aanm reviewed (changes requested):
Can you add more information in the 2nd commit why is it needed? it's not clear to me why would clustermesh would need to know about that option.

christarazi commented on pkg/k8s/watchers/watcher.go:
```diff
@@ -267,9 +267,9 @@ func (k *K8sWatcher) WaitForCRDsToRegister(ctx context.Context) error {
 // To be called after WaitForCRDsToRegister() so that all needed CRDs have
 // already been registered.
-func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context) <-chan struct{} {
+func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context, cfg WatcherConfiguration) <-chan struct{} {
```
Instead of passing in `cfg` to each function that may use it, we could add it as a field to the `K8sWatcher`

christarazi commented on operator/watchers/k8s_service_sync.go:
```diff
@@ -91,11 +90,17 @@ func k8sServiceHandler() {
 
+// ServiceSyncConfiguration is the required configuration for StartSynchronizingServices
+type ServiceSyncConfiguration interface {
+	// LocalClusterName must return the local cluster name`
```
Nit: extra backtick at the end of the godoc

christarazi reviewed (changes requested):
One comment on adding a field to an existing struct instead of expanding the set of parameters passed into functions and another small nit

tgraf committed:
clustermesh-apiserver: Support K8sServiceProxyName

The K8sServiceProxyName affects the subset of services Cilium should
implement and thus must be respected by ClusterMesh as well.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
clustermesh-apiserver: Correctly set the cluster name

The package pkg/service/store is consumed by the clustermesh-apiserver.
It depends on pkg/option internally. Due to not setting up
option.Config.ClusterName correctly, default values have been used.
Refactor pkg/service/store to no longer depend on pkg/option and require
users to pass in the required configuration settings via an interface.

Signed-off-by: Thomas Graf <thomas@cilium.io>
